packages/table/src/common/grid.ts

```diff
@@ -321,10 +329,17 @@ export class Grid {
             }
 
             case RegionCardinality.FULL_COLUMNS: {
-                const cellRect0 = this.getCellRect(0, region.cols[0]);
-                const cellRect1 = this.getCellRect(0, region.cols[1]);
+                const [colStart, colEnd] = region.cols;
+
+                // if the region is outside the bounds of the table, don't display it
+                if (this.isGhostIndex(0, colStart)
+                    || this.isGhostIndex(0, colEnd)) {
```

Comment:
Chatted with Gilad. Would probably just put these checks on one line to avoid the holy war over how to properly format this. 😛